src/main/java/gyro/google/iam/ServiceAccountResource.java

```diff
@@ -0,0 +1,254 @@
+    @Required
+    @Regex(value = "^[a-z]([a-z]|[0-9]|-){5,29}$", message = "a string starting with a lowercase letter followed by lower case alphanumerical characters or hyphens and between 6 and 30 characters.")
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Enable or disable the service account.
+     */
+    @Updatable
+    public Boolean getEnableAccount() {
+        return enableAccount;
+    }
+
+    public void setEnableAccount(Boolean enableAccount) {
+        this.enableAccount = enableAccount;
+    }
+
+    /**
+     * The ID of the service account.
+     */
+    @Id
+    @Output
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * The email address of the service account.
+     */
+    @Output
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    @Override
+    public void copyFrom(ServiceAccount model) throws Exception {
+        setId(model.getName());
+        setEmail(model.getEmail());
+        setName(Utils.getServiceAccountNameFromId(model.getName()));
+        setDescription(model.getDescription());
+        setDisplayName(model.getDisplayName());
+        setEnableAccount((model.getDisabled() == null || model.getDisabled().equals(Boolean.FALSE)) ? Boolean.TRUE : Boolean.FALSE);
+    }
+
+    @Override
+    public String primaryKey() {
+        return super.primaryKey();
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        Iam client = createClient(Iam.class);
+
+        List<ServiceAccount> serviceAccounts = client.projects()
+            .serviceAccounts()
+            .list("projects/" + getProjectId())
+            .execute()
+            .getAccounts()
+            .stream()
+            .filter(r -> r.getName().equals(getId()))
+            .collect(
+                Collectors.toList());
+
+        if (serviceAccounts.isEmpty()) {
+            return false;
+        }
+
+        copyFrom(serviceAccounts.get(0));
+
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        Iam client = createClient(Iam.class);
+
+        ServiceAccount serviceAccount = new ServiceAccount();
+        serviceAccount.setDisplayName(getDisplayName());
+
+        if (getDescription() != null) {
+            serviceAccount.setDescription(getDescription());
+        }
+
+        CreateServiceAccountRequest request = new CreateServiceAccountRequest();
+        request.setAccountId(getName());
+        request.setServiceAccount(serviceAccount);
+
+        ServiceAccount response = client.projects()
+            .serviceAccounts()
+            .create("projects/" + getProjectId(), request)
+            .execute();
+
+        setId(response.getName());
```

Comment:
So that if it fails in the next api call, corresponding gyro up can acknowledge that the resource was created
```suggestion
        setId(response.getName());
        state.save();
```